src/utils/jsonPointers.ts

```diff
@@ -56,23 +84,28 @@ export const jsonPointerForPosition = (
  */
 export const getJsonPointers = (
   state: EditorState,
-  mode: JSONMode = "json4"
+  mode: JSONMode
 ): JSONPointersMap => {
-  const json = syntaxTree(state);
+  const tree = syntaxTree(state);
   const pointers: JSONPointersMap = new Map();
-  json.iterate({
+  tree.iterate({
     enter: (type: SyntaxNodeRef) => {
-      if (type.name === "PropertyName" || type.name === "Object") {
-        const pointer = getJsonPointerAt(state.doc, type.node);
+      if (
+        [TOKENS.PROPERTY_NAME, TOKENS.OBJECT].includes(
+          resolveTokenName(type.name, mode) as any
+        )
+      ) {
+        const pointer = getJsonPointerAt(state.doc, type.node, mode);
 
         const { from: keyFrom, to: keyTo } = type.node;
         // if there's no value, we can't get the valueFrom/to
         if (!type.node?.nextSibling?.node) {
           pointers.set(pointer, { keyFrom, keyTo });
           return true;
         }
+        // TODO: Make this generic enough to avoid mode-specific checks
```

Comment:
just adding a note from a deep dive on this

```suggestion
        // TODO: Make this generic enough to avoid mode-specific checks.
        // It appears that in the official json mode grammar, the right hand colon
        // after the property name is not given a node, or perhaps it was the other way around.
```